src/Modal.js

```diff
@@ -161,7 +161,11 @@ const Modal = React.createClass({
     let show = !!props.show;
     let dialog = React.Children.only(this.props.children);
 
-    let setMountNode = ref => this.mountNode = (!ref || ref.getMountNode());
+    let setMountNode = ref => {
+      if (ref !== null && typeof ref === 'object') {
```

Comment:
also, should probably highlight this PR is a bit confusing as multiple modals are not a problem per se, I fixed this in our production code by weeding out unnecessary renders with `shouldComponentUpdate`.